src/lib/components/ConfirmTransactions.js

```diff
@@ -12,78 +14,189 @@ ${json}
-                </span>
+                <div>
+                  <span className="text-secondary">Method name: </span>
+                  <span className="font-monospace">{transaction.methodName}</span>
+                </div>
+                {transaction.deposit && transaction.deposit.gt(0) && (
+                  <div>
+                    <span className="text-secondary">Deposit: </span>
+                    <span className="font-monospace">
+                      {displayNear(transaction.deposit)}
+                    </span>
+                  </div>
+                )}
+                <div>
+                  <span className="text-secondary">Gas: </span>
+                  <span className="font-monospace">
+                    {displayGas(transaction.gas)}
+                  </span>
+                </div>
+                <Markdown text={jsonMarkdown(transaction.args)} />
               </div>
-              <Markdown text={jsonMarkdown(transaction.args)} />
+            ))}
+          {transactions.length == 1 ?
+            <div class="form-check form-switch">
+              <input class="form-check-input" type="checkbox" id="dontaskagaincheckbox"
+                checked={dontAskAgainChecked}
+                onChange={() => dontAskAgainCheckboxChange()}
+              />
+              <label class="form-check-label" for="dontaskagaincheckbox">
+                Don't ask again for sending similar transactions by{" "}
+                <span className="font-monospace">{widgetSrc}</span>
+              </label>
             </div>
-          ))}
-      </Modal.Body>
-      <Modal.Footer>
-        <button
-          className="btn btn-success"
-          disabled={loading}
-          onClick={(e) => {
-            e.preventDefault();
-            setLoading(true);
-            near.sendTransactions(transactions).then(() => {
+            :
+            <></>
+          }
+        </Modal.Body>
+        <Modal.Footer>
+          <button
+            className="btn btn-success"
+            disabled={loading}
+            onClick={async (e) => {
+              e.preventDefault();
+              setLoading(true);
+              if (dontAskAgainChecked) {
+                const contractId = transactions[0].contractName;
+                const methodName = transactions[0].methodName;
+                const permissionObject = (await getWidgetContractPermission(widgetSrc, contractId)) || {};
+                permissionObject[methodName] = true;
+
+                cache.localStorageSet(
+                  StorageDomain,
+                  {
+                    widgetSrc,
+                    contractId,
+                    type: StorageType.SendTransactionWithoutConfirmation,
+                  },
+                  permissionObject
+                );
+
+                if (!(await near.isSignedIntoContract(contractId))) {
+                  near.signInAndSetPendingTransaction(transactions[0]);
+                  return;
```

Comment:
I feel that `setLoading(false)` and `onHide` needs to be called here to work fine with browser-extension wallets (e.g. Sender)